core/metacling/src/TClingDataMemberInfo.cxx

```diff
@@ -528,6 +530,7 @@ const char *TClingDataMemberInfo::TypeName() const
    CheckForIoTypeAndName();
    if (!fIoType.empty()) return fIoType.c_str();
 
+   cling::Interpreter::PushTransactionRAII RAII(fInterp);
```

Comment:
Please re-check the scope of the RAII guards; they should be close to where the deserialization / instantiation actually happens. In this case likely within the `if` below?